src/app/backend/namespacelist.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+// Returns a list of all namespaces in the cluster.
+func GetNamespaceList(client *client.Client) ([]string, error) {
```

Comment:
I agree, I have simplified it too much. Previous version with one structure inside another also was not good. I will do it like you said, JSON object returned by API will contain list of strings (namespaces) and we will be able to add another properties like timestamp etc. in the future.